### PR TITLE
Allow us to turn security on/off at will

### PR DESCRIPTION
**Pull request by mlaccetti**
And if turned on, choose a different user/group

Fixed the formatting of the `preferred` affinity to be indented to the correct level.

### CODE REVIEW

**Comment by mlaccetti** [comment]
@k8s-ci-robot CLA signed.

For real, this time. I swear.

File: incubator/zookeeper/templates/ss.yaml [hunk 1]
```diff
@@ -100,9 +100,12 @@ spec:
         volumeMounts:
         - name: datadir
           mountPath: /var/lib/zookeeper
+          subPath: data
+      {{- if eq .Values.Security.Enabled true}}
```
Comment (fhemberger) [inline incubator/zookeeper/templates/ss.yaml]: Please fix whitespace `{{- if eq .Values.Security.Enabled true }}`

File: incubator/zookeeper/templates/ss.yaml [hunk 2]
```diff
@@ -100,9 +100,12 @@ spec:
...
       securityContext:
-        runAsUser: 1000
-        fsGroup: 1000
+        runAsUser: {{.Values.Security.RunAsUser}}
```
Comment (fhemberger) [inline incubator/zookeeper/templates/ss.yaml]: Please fix whitespace `{{ .Values.Security.RunAsUser }}` and `{{ .Values.Security.FsGroup }}`

File: incubator/zookeeper/templates/ss.yaml [hunk 3]
```diff
@@ -31,15 +31,15 @@ spec:
       {{- else if eq .Values.AntiAffinity "soft"}}
```
Comment (fhemberger) [inline incubator/zookeeper/templates/ss.yaml]: Please fix whitespace `{{- else if eq .Values.AntiAffinity "soft" }}`

File: incubator/zookeeper/templates/ss.yaml [hunk 4]
```diff
@@ -100,9 +100,12 @@ spec:
...
         - name: datadir
           mountPath: /var/lib/zookeeper
+          subPath: data
+      {{- if eq .Values.Security.Enabled true}}
       securityContext:
-        runAsUser: 1000
-        fsGroup: 1000
+        runAsUser: {{.Values.Security.RunAsUser}}
+        fsGroup: {{.Values.Security.FsGroup}}
+      {{- end}}
```
Comment (fhemberger) [inline incubator/zookeeper/templates/ss.yaml]: Please fix whitespace `{{- end }}`

**Review by fhemberger — CHANGES_REQUESTED** [review]

**Comment by mlaccetti** [comment]
@fhemberger Fixed all of the spacing issues, to match the desired format.

**Comment by fhemberger** [comment]
@mlaccetti Cool, thanks!

**Comment by viglesiasce** [comment]
@mlaccetti @fhemberger not sure whats going on but way too many files have changed in this PR. Can you see if you can fix it?

**Comment by fhemberger** [comment]
@mlaccetti Oh, I only meant to fix the whitespace for your zookeeper changes, not for the entire repo (that change has already been merged).

**Comment by unguiculus** [comment]
@fhemberger Can you please fix this PR so only Zookeeper-related changes are in?

**Comment by fhemberger** [comment]
@unguiculus Oh, I would if it was my PR. 😀  I only suggested changes.
@mlaccetti Can you please check the PR again?

**Comment by unguiculus** [comment]
@fhemberger Of course, sorry, I was confused.

**Comment by mlaccetti** [comment]
@fhemberger @unguiculus Let me figure out where all the extra changes came from and address that.

**Comment by mlaccetti** [comment]
@unguiculus The changes were there since my branch was out of date.  Fixed, and we're back to just the two changes. :)

**Comment by mlaccetti** [comment]
@unguiculus looks like the automated deployment isn't happy - might be due to PVCs not being provisioned, etc. How do you want to handle this?

**Comment by unguiculus** [comment]
Sorry for the delay, I was on vacation. Can you fix the conflicts, please?

**Comment by mlaccetti** [comment]
Ugh rebase went to shit.  Let me try to fix.

**Comment by unguiculus** [comment]
Marking this as stale. Please update PR within one week. Also, note that there is a problem with the CLA.

**Comment by unguiculus** [comment]
I fixed the history for you.

File: incubator/zookeeper/templates/statefulset.yaml [hunk 5]
```diff
@@ -94,9 +94,12 @@ spec:
         volumeMounts:
         - name: datadir
           mountPath: /var/lib/zookeeper
+          subPath: data
```
Comment (unguiculus) [inline incubator/zookeeper/templates/statefulset.yaml:97]: How is the `subPath` related to your change? Does that make sense?

**Review by unguiculus — CHANGES_REQUESTED** [review]

File: incubator/zookeeper/templates/statefulset.yaml (hunk 5 above)
Comment (mlaccetti) [reply incubator/zookeeper/templates/statefulset.yaml:97]: When AWS mounts EBS as a PVC, it has some "stuff" already in there, and ZK loves to die saying that the directory is already in use.  Put in a subpath, and it'll be nice and clean, and no more complaints.

**Comment by unguiculus** [comment]
Reset author.

**Review by unguiculus — APPROVED** [review]

**Pull request merged**